pkg/pipeline/transform/network_services/netdb.go

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright (C) 2022 IBM, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * > Note: this code is a revised and enhanced version of the netdb.go file
+ * > from https://github.com/dominikh/go-netdb/ (MIT License)
+ */
+
+package netdb
+
+import (
+	"io/ioutil"
+	"strconv"
+	"strings"
+)
+
+type Protoent struct {
+	Name    string
+	Aliases []string
+	Number  int
+}
+
+type Servent struct {
+	Name     string
+	Aliases  []string
+	Port     int
+	Protocol *Protoent
+}
+
+// These variables get populated from /etc/protocols and /etc/services
+// respectively.
+var (
+	Protocols []*Protoent
+	Services  []*Servent
+)
+
+func Init(protocolsFile, servicesFile string) error {
+	Protocols = []*Protoent{}
+	Services = []*Servent{}
+
+	protoMap := make(map[string]*Protoent)
+
+	// Load protocols
+	if protocolsFile == "" {
+		protocolsFile = "/etc/protocols"
+	}
+	data, err := ioutil.ReadFile(protocolsFile)
+	if err != nil {
+		return err
+	}
+
+	for _, line := range strings.Split(string(data), "\n") {
+		line = strings.TrimSpace(line)
+		split := strings.SplitN(line, "#", 2)
+		fields := strings.Fields(split[0])
+		if len(fields) < 2 {
+			continue
+		}
+
+		num, err := strconv.ParseInt(fields[1], 10, 32)
+		if err != nil {
+			return err
+		}
+
+		protoent := &Protoent{
+			Name:    fields[0],
+			Aliases: fields[2:],
+			Number:  int(num),
+		}
+		Protocols = append(Protocols, protoent)
+
+		protoMap[fields[0]] = protoent
+	}
+
+	// Load services
+	if servicesFile == "" {
+		servicesFile = "/etc/services"
+	}
+	data, err = ioutil.ReadFile(servicesFile)
+	if err != nil {
+		return err
+	}
+
+	for _, line := range strings.Split(string(data), "\n") {
+		line = strings.TrimSpace(line)
+		split := strings.SplitN(line, "#", 2)
+		fields := strings.Fields(split[0])
+		if len(fields) < 2 {
+			continue
+		}
+
+		name := fields[0]
+		portproto := strings.SplitN(fields[1], "/", 2)
+		port, err := strconv.ParseInt(portproto[0], 10, 32)
+		if err != nil {
+			return err
+		}
+
+		proto := portproto[1]
+		aliases := fields[2:]
```

Comment:
Is this OK if `fields` only has 2 entries i.e. if there is no alias specified for the service?

Reply:
Sure ... look on the following examples::

This is a line with no aliases::
`med-ci          24005/tcp               # med-ci`

This is a line **with** aliases::
`iceedcp_rx      31949/udp  iceedcp-rx   # Embedded Device Configuration Protocol RX` 

They are all parsed correctly as part of the `/etc/services` file